generator/scripts/generateInputRange.py

```diff
@@ -0,0 +1,97 @@
+import openpyxl  # Read data from Excel
+
+
+# Function to replace single quotes and stringify text
+def replaces_quotes_and_stringify(text):
+    if text is not None:
+        return str(text).replace("'", "\\'")  # Replace single quotes
+    return None
+
+
+# Function to generate inputRange.tsx
+def generate_input_range(input_file: str, output_file: str):
```

Comment:
Here you can split this function into a few parts:

1- Load the file, which is now an excel but can eventually be a csv or anything else
2- data sanitization (the part under your comment #Getvalues from the row dictionary)
3- The typescript generator itself, which takes the data and generates the code

If you look at the fillLocales.py script in Evolution, which has separate classes for reading input file and writing output data. Small classes and/or functions with clear responsibilities are to be favored

Reply:
Yes I will try to separate them better, thank you.